src/uct/base/uct_iface.c

```diff
@@ -732,3 +816,67 @@ uct_ep_keepalive_check(uct_ep_h tl_ep, uct_keepalive_info_t *ka, unsigned flags,
 
     return UCS_OK;
 }
+
+static void
+uct_iface_vfs_show_cap(void *obj, void *arg, ucs_string_buffer_t *strb)
+{
+    ucs_string_buffer_appendf(strb, "1\n");
+}
+
+static void
+uct_iface_vfs_show_cap_limit(void *obj, void *arg, ucs_string_buffer_t *strb)
+{
+    uct_iface_h iface = obj;
+    uct_iface_attr_t iface_attr;
+    size_t attr;
+
+    if (uct_iface_query(iface, &iface_attr) != UCS_OK) {
+        ucs_string_buffer_appendf(strb, "<failed to query iface attributes>\n");
+        return;
+    }
+    attr = *(size_t*)UCS_PTR_BYTE_OFFSET(&iface_attr, *(size_t*)(arg));
+    ucs_string_buffer_appendf(strb, "%zu\n", attr);
+}
+
+static void uct_iface_vfs_init_caps(uct_iface_h iface, uint64_t iface_cap_flags)
+{
+    size_t i;
+
+    for (i = 0; i < ucs_static_array_size(uct_iface_cap_infos); ++i) {
+        if (iface_cap_flags & uct_iface_cap_infos[i].flag) {
+            ucs_vfs_obj_add_ro_file(iface, uct_iface_vfs_show_cap, NULL,
+                                    "attribute/capability/%s",
+                                    uct_iface_cap_infos[i].name);
+        }
+    }
+}
+
+static void
+uct_iface_vfs_init_cap_limits(uct_iface_h iface, uint64_t iface_cap_flags)
+{
+    size_t i;
+
+    for (i = 0; i < ucs_static_array_size(uct_iface_cap_limit_infos); ++i) {
+        if (iface_cap_flags & uct_iface_cap_limit_infos[i].flag) {
+            ucs_vfs_obj_add_ro_file(iface, uct_iface_vfs_show_cap_limit,
+                                    (void*)&uct_iface_cap_limit_infos[i].offset,
+                                    "attribute/%s/%s",
+                                    uct_iface_cap_limit_infos[i].op_name,
+                                    uct_iface_cap_limit_infos[i].limit_name);
+        }
+    }
+}
+
+void uct_iface_vfs_refresh(void *obj)
+{
+    uct_iface_h iface = obj;
+    uct_iface_attr_t iface_attr;
+
+    if (uct_iface_query(iface, &iface_attr) != UCS_OK) {
+        ucs_error("Failed to query iface attributes");
```

Comment:
1. smallcase
2. better not print ucs_error here. maybe ucs_debug, or add file like in line 804?